com.unity.render-pipelines.core/Tests/Editor/ContextualMenuDispatcherTests.cs

```diff
@@ -0,0 +1,86 @@
+using NUnit.Framework;
+using System;
+using UnityEditor.Rendering;
+
+namespace UnityEngine.Rendering.Tests
+{
+    #region Components
+    class Apple : MonoBehaviour
+    {
+
+    }
+
+    [RequireComponent(typeof(Apple))]
+    class AdditionalApple : MonoBehaviour, IAdditionalData
+    {
+    }
+
+    class Banana : MonoBehaviour
+    {
+    }
+    [RequireComponent(typeof(Banana))]
+    class AdditionalBanana : MonoBehaviour, IAdditionalData
+    {
+    }
+
+    [RequireComponent(typeof(Apple))]
+    [RequireComponent(typeof(Banana))]
+    class FruitBasket : MonoBehaviour, IAdditionalData
+    {
+    }
+    #endregion
+
+    class ContextualMenuDispatcherTests
+    {
```

Comment:
As an example of property testing

```suggestion
    {
        static class Properties
        {
            public static bool RemoveComponentRemovesOnlyComponentAndDependencies(Type componentType, Type[] componentsInGameObject)
            {
                GameObject gameObject = new GameObject("Test");
                try
                {
                    // Add all components in the gameObject
                    
                    // Find all component types that are expected to be deleted
                    HashSet<Type> expectedDeletedComponents;
                    
                    var component = gameObject.GetComponent(componentType);
                    ContextualMenuDispatcher.RemoveComponent(component);
                    
                    // Check only expected components are deleted, return false if not
                    
                    return true;
                }
                finalize
                {
                    GameObject.DestroyImmediate(gameObject);
                }
            }
        }
        
        static Type[] k_AllComponentTypes = new Type[] { typeof(Apple), typeof(Banana), typeof(FruitBasket) };
        static Type[] k_ComponentTypes = new Type[] { typeof(Apple), typeof(Banana) };
        static Type[][] k_ComponentTypeArraysForGameObject = Enumerable.Range(0, 5)
            // Initialize Random with a fixed state
            .Select(i => { if (i == 0) {Random.InitState(98540329);} return i;})
            // Randomly choose components
            .Select(_ => Enumerable.Range(0, k_AllComponentTypes .Length - 1)
                .Select(_ => k_AllComponentTypes[Random.Range(0, k_AllComponentTypes.Length - 1)])
                .Distinct()
                .ToArray();
        
        [Test]
        public static void PropertyRemoveComponentRemovesOnlyComponentAndDependencies(
            [ValuesSource(nameof(k_ComponentTypes)] Type componentType,
            [ValuesSource(nameof(k_ComponentTypeArraysForGameObject)] Type[] gameObjectComponents
         ) => Assert.True(Properties.RemoveComponentRemovesOnlyComponentAndDependencies(componentType, gameObjectComponents);
    
```

Reply:
seems a little overkill in this context. there is not a lot of possibilities so I believe a set of test case would be enough here.

Case worth testing: